lib/index.js

```diff
@@ -228,15 +231,17 @@ class Service extends AdapterService {
     // By default we will just query for the one id. For multi patch
     // we create a list of the ids of all items that will be changed
     // to re-query them after the update
-    const ids = id === null ? this._getOrFind(null, params)
-      .then(mapIds) : Promise.resolve([id]);
+    const ids = this._getOrFind(id, params).then(mapIds);
 
     return ids.then(idList => {
       // Create a new query that re-queries all ids that
       // were originally changed
-      const findParams = Object.assign({}, params, Object.assign({}, {
-        query: Object.assign({ [this.id]: { $in: idList } }, params.query)
-      }));
+      const findQuery = Object.assign(
+        { [this.id]: { $in: idList } },
+        _.pick(params.query || {}, '$select', '$sort', '$limit', '$skip')
```

Comment:
We can use `filterQuery` (https://github.com/feathersjs/databases/blob/master/packages/adapter-commons/src/filter-query.ts#L102) here:

```js
const { query } = filterQuery(params.query);
```

Reply:
This has been updated. I believe you were actually recommending `this.filterQuery(params).filters` correct?